analysis-component/src/main/java/org/hobbit/analysis/AnalysisComponent.java

```diff
@@ -81,121 +89,149 @@ public void run() throws Exception {
+                    LOGGER.info("Calculating prediction model...");
+                    model.predictPerformanceOfSystem();
 
-                            LOGGER.info("Enhancing experiment model with results...");
-                            model.enhanceExperimentModel();
-                            updatedModel = model.getUpdatedModel();
-                        }
-                        catch (Exception e){
-                            LOGGER.error("Error in analyzing data. Have to abort analysis...", e);
-                        }
-                    }
-                } catch (Exception e) {
-                    LOGGER.error("Error in pre-processing. ", e);
+                    LOGGER.info("Computing correlation...");
+                    model.computeCorrelation(benchmarkUri, systemUri);
+
+                    LOGGER.info("Analysis performed successfully!");
+
+                    LOGGER.info("Enhancing experiment model with results...");
+                    model.enhanceExperimentModel();
+                    updatedModel = model.getUpdatedModel();
                 }
-                if (updatedModel != null) {
-                    try {
-                        LOGGER.info("Updating model...");
-                        String sparqlUpdateQuery = null;
-                        //TODO:: handle null exception for sparql queries
-                        sparqlUpdateQuery = SparqlQueries.getUpdateQueryFromDiff(experimentModel,
-                                updatedModel,
-                                GRAPH_URI);
-                        LOGGER.info("Sending the enhanced model to storage...");
-                        storage.sendUpdateQuery(sparqlUpdateQuery);
-                    } catch (Exception e) {
-                        LOGGER.error("Error when updating model.", e);
-                    }
-                } else {
-                    LOGGER.error("Model did not update properly! No result model from the analysis.");
+                catch (Exception e){
+                    LOGGER.error("Error in analyzing data. Have to abort analysis...", e);
                 }
             }
+        } catch (Exception e) {
+            LOGGER.error("Error in pre-processing. ", e);
+        }
+        if (updatedModel != null) {
+            try {
+                LOGGER.info("Updating model...");
+
+                // Remove existing analysis results for this pair of benchmark and system instance.
+                Resource resultset = updatedModel.listResourcesWithProperty(RDF.type, HOBBIT.AnalysisResultset).next();
+                storage.sendUpdateQuery(SparqlQueries.deleteAnalysisResults(resultset.getURI(), Constants.PUBLIC_RESULT_GRAPH_URI));
+
+                String sparqlUpdateQuery = null;
+                //TODO:: handle null exception for sparql queries
+                sparqlUpdateQuery = SparqlQueries.getUpdateQueryFromDiff(experimentModel,
+                        updatedModel,
+                        GRAPH_URI);
```

Comment:
Please make sure that the original model is replaced with `null` and the updated model is initialized as empty model.